lib/svtplay_dl/postprocess/__init__.py

```diff
@@ -75,125 +79,268 @@ def query(self):
             subfile = "{0}.srt".format(os.path.splitext(formatname(self.stream.output, self.config,
                                                                    self.stream.output_extention))[0])
-            langs += [query(subfile)]
+            lang['lang'] = query(subfile)
+            lang['sub_file'] = subfile
+            lang['title'] = None
+            langs.append(lang)
+
         if len(langs) >= 2:
-            logging.info("Language codes: " + ', '.join(langs))
+            lang_codes = set([l['lang'] for l in langs])
+            logging.info("Language codes: " + ', '.join(lang_codes))
         else:
-            logging.info("Language code: " + langs[0])
+            logging.info("Language code: " + langs[0]['lang'])
         return langs
 
-    def remux(self):
-        if self.detect is None:
-            logging.error("Cant detect ffmpeg or avconv. Cant mux files without it.")
+    def remux_mkv(self):
+        # remux to mkv, mkvmerge prefered over ffmpeg
+        # todo better error message/checking
+        if self.detect_mkvmerge:
+            #  TODO add something similar to this to allow config to overwrite priority
+            #  and not self.config.get('prefered_mkv_tool', 'mkvmerge') == 'ffmpeg':
+            self.remux_mkv_mkvmerge()
+        else:
+            logging.debug("Can't detect mkvmerge, using ffmpeg/avconv instead")
+            self.remux_mkv_ffmpeg()  # no check needed here as the function will throw an error if ffmpeg/avconv not found
+
+    def remux_mkv_ffmpeg(self):
+        # mux to mkv with ffmpeg
+        if self.detect_ffmpeg is None:
+            logging.error("Cant detect ffmpeg or avconv. Can't mux files without it.")
             return
         if self.stream.finished is False:
             return
 
-        if formatname(self.stream.output, self.config, self.stream.output_extention).endswith('.mp4') is False:
-            orig_filename = formatname(self.stream.output, self.config, self.stream.output_extention)
-            name, ext = os.path.splitext(orig_filename)
-            new_name = u"{0}.mp4".format(name)
+        self.mkv = True
+        langs = []
+        orig_filename = formatname(self.stream.output, self.config, self.stream.output_extention)
+        name, ext = os.path.splitext(orig_filename)
+        new_name = orig_filename
+        if formatname(self.stream.output, self.config, self.stream.output_extention).endswith('.mkv') is False:
+            new_name = u"{0}.mkv".format(name)
 
-            cmd = [self.detect, "-i", orig_filename]
-            _, stdout, stderr = run_program(cmd, False)  # return 1 is good here.
-            videotrack, audiotrack = self._checktracks(stderr)
+        # First remux to mp4 if .ts file to change bitstream of audio tracks
+        # this is technically only needed if we do not have a new version of ffmpeg it works with >3.0 at least.
+        # TODO figure out which version and add check to only run this if needed
```

Comment:
i think we need to support only 3.0 or what version of ffmpeg we need with tv4play.

Reply:
In that case I have removed the code:)